src/js/components/TaskFileListComponent.jsx

```diff
@@ -0,0 +1,179 @@
+var React = require("react/addons");
+var classNames = require("classnames");
+var lazy = require("lazy.js");
+
+var Util = require("../helpers/Util");
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+var MesosStore = require("../stores/MesosStore");
+
+var TaskFileListComponent = React.createClass({
+  displayName: "TaskFileListComponent",
+  propTypes: {
+    task: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    var task = this.props.task;
+    var taskId = task.id || task.taskId;
```

Comment:
I only added those checks is they were implemented within the [`TaskMesosUrlComponent`](https://github.com/mesosphere/marathon-ui/blob/master/src/js/components/TaskMesosUrlComponent.jsx#L55) and I wanted to be safe... but yeah, let's move that check to the `AppsStore``